releases/describe_signed_firmware.py

```diff
@@ -45,38 +45,47 @@ def main() -> int:
         return 1
 
     with open(filename, "rb") as fileobj:
-        signed_firmware = fileobj.read()
+        file_content = fileobj.read()
 
-    # Split signed firmware into sigdata and firmware
-    if len(signed_firmware) < SIGDATA_LEN:
-        print("firmware too small")
-        return 1
-    magic, rest = signed_firmware[:MAGIC_LEN], signed_firmware[MAGIC_LEN:]
-    sigdata, firmware = rest[:SIGDATA_LEN], rest[SIGDATA_LEN:]
+    firmware_is_signed = len(file_content) >= MAGIC_LEN and file_content[:MAGIC_LEN] in [MAGIC_MULTI, MAGIC_BTCONLY]
 
-    print(
-        "The following information assumes the provided binary was signed correctly; "
-        "the signatures are not being verified."
-    )
-    if magic == MAGIC_MULTI:
-        print("This is a Multi-edition firmware.")
-    elif magic == MAGIC_BTCONLY:
-        print("This is a Bitcoin-only edition firmware.")
+    if firmware_is_signed:
```

Comment:
the file is named `describe_signed_firmware.py` and its purpose is to check signed firmwares not unsigned ones.

To keep its focus narrow and this change simple, it would be better to exit with an error if the magic is not recognized, i.e. instead of printing

> Unrecognized firmware edition; magic = c8f10120

and then continuing, it could just stop after that error. Maybe expand this message with something like "This tool does not support unsigned firmware. You can use `sha256sum` directly on unsigned firmware binaries to get its hash" or similar.